client/sites-dashboard/hooks/use-site-favorite-mutation.ts

```diff
@@ -0,0 +1,69 @@
+import { useMutation, UseMutationOptions, useIsMutating } from '@tanstack/react-query';
+import wp from 'calypso/lib/wp';
+
+interface MutationResponse {
+	message: string;
+}
+
+interface MutationError {
+	code: string;
+	message: string;
+}
+
+export const ADD_SITE_FAVORITE = 'add-site-favorite-key';
+export const DELETE_SITE_FAVORITE = 'delete-site-favorite-key';
+
+export const useAddSiteFavorite = (
+	siteId: number,
+	options: UseMutationOptions< MutationResponse, MutationError >
+) => {
+	const mutation = useMutation( {
+		mutationFn: async () =>
+			wp.req.post( {
+				path: `/sites/${ siteId }/site-favorite`,
+				apiNamespace: 'wpcom/v2',
+			} ),
+		...options,
+		mutationKey: [ ADD_SITE_FAVORITE, siteId ],
+		onSuccess: async ( ...args ) => {
+			options.onSuccess?.( ...args );
```

Comment:
nit: `onSuccess` could be removed if you don't do anything else in here except for calling `options.onSuccess`.